synapse/rest/client/v2_alpha/register.py

```diff
@@ -236,7 +237,7 @@ def on_POST(self, request):
             add_email = True
 
         result = yield self._create_registration_details(
-            registered_user_id
+            registered_user_id, body
```

Comment:
Right, yes. I think the way I've implemented this in the js-sdk (https://github.com/matrix-org/matrix-js-sdk/pull/151) means it will actually send a device_id in the follow-up request, but ISWYM.

I think I want to add some sytests to this though, so that I know what I'm doing.